### PR TITLE
Deployment restructure

### PR DESCRIPTION
**Pull request by thehonker**
This PR is to mainline our production deployment changes
It introduces several breaking changes to layout and deployment:

- Restructures the compose stack to have independent mariadb and nginx
- Assumes that it is behind an external traefik
- Supplies the DB hostname via envvars instead of hardcoded
- Sets up autobuilds via actions/ghcr

### CODE REVIEW

File: .github/workflows/docker-autobuild.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v2
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: ./forum/
```
Comment (itsrachelfish) [inline .github/workflows/docker-autobuild.yml]: Why is the forum context being used here?

**Review by itsrachelfish — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-autobuild.yml (hunk 1 above)
Comment (thehonker) [reply .github/workflows/docker-autobuild.yml]: Must be a ^c^v overlook, it should be just `context: .`

Comment (thehonker) [reply .github/workflows/docker-autobuild.yml]: Fixed in [https://github.com/wetfish/danger/pull/1/commits/a70027eb659dde669d61d30bb35bb46a325c1796](https://github.com/wetfish/danger/pull/1/commits/a70027eb659dde669d61d30bb35bb46a325c1796)

**Review by itsrachelfish — APPROVED** [review]
Tested locally, working as expected :tada: 

image

**Pull request merged**